crates/oxc_query/src/properties.rs

```diff
@@ -0,0 +1,584 @@
+        _ => {
+            unreachable!(
+                "attempted to read unexpected property '{property_name}' on type 'SimpleExtend'"
+            )
+        }
+    }
+}
+
+pub(super) fn resolve_span_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "end" => resolve_property_with(contexts, |v| {
+            v.as_span()
+                .unwrap_or_else(|| panic!("expected to have a span vertex, instead have: {v:#?}"))
+                .end
+                .into()
+        }),
+        "start" => resolve_property_with(contexts, |v| {
+            v.as_span()
+                .unwrap_or_else(|| panic!("expected to have a span vertex, instead have: {v:#?}"))
+                .start
+                .into()
+        }),
+        _ => {
+            unreachable!("attempted to read unexpected property '{property_name}' on type 'Span'")
+        }
+    }
+}
+
+pub(super) fn resolve_specific_import_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "local_name" => resolve_property_with(contexts, |v| {
+            v.as_specific_import()
+                .unwrap_or_else(|| {
+                    panic!("expected to have a specificimport vertex, instead have: {v:#?}")
+                })
+                .local
+                .name
+                .to_string()
+                .into()
+        }),
+        "original_name" => resolve_property_with(contexts, |v| {
+            v.as_specific_import()
+                .unwrap_or_else(|| {
+                    panic!("expected to have a specificimport vertex, instead have: {v:#?}")
+                })
+                .imported
+                .name()
+                .to_string()
+                .into()
+        }),
+        _ => {
+            unreachable!(
+                "attempted to read unexpected property '{property_name}' on type 'SpecificImport'"
+            )
+        }
+    }
+}
+
+pub(super) fn resolve_type_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+    adapter: &'a Adapter<'b>,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "str" => resolve_property_with(contexts, |v| {
+            let span = v
+                .as_type()
+                .unwrap_or_else(|| panic!("expected to have a type vertex, instead have: {v:#?}"))
+                .span();
+            adapter.semantic.source_text()[span.start as usize..span.end as usize].into()
+        }),
+        _ => {
+            unreachable!("attempted to read unexpected property '{property_name}' on type 'Type_'")
```

Comment:
did you mean `Type` here? is this an artifact of the escaping you temporarily introduced to make stubgen work?
```suggestion
            unreachable!("attempted to read unexpected property '{property_name}' on type 'Type'")
```